src/shutdown/umount.c

```diff
@@ -716,10 +739,53 @@ static int mount_points_list_umount(MountPoint **head, bool *changed, bool last_
                         continue;
 
                 /* Trying to umount */
-                if (umount_with_timeout(m, last_try) < 0)
+                r = umount_with_timeout(m, last_try);
+                if (r < 0)
                         n_failed++;
                 else
                         *changed = true;
+
+                /* If a mount is busy, we move it to not keep parent mount points busy.
+                 * If a mount point is not a leaf, moving it would invalidate our mount table.
+                 * More moving will occur in next iteration with a fresh mount table.
+                 *
+                 * umount_with_timeout will return EPROTO even on EBUSY due to forking.
+                 */
+                if (r == -EPROTO && m->leaf) {
```

Comment:
Maybe invert this to reduce indentation of the remaining code?
```c
if (r != -EPROTO || !m->leaf)
       continue;
```
?

Reply:
Sure, that looks better. Done.